aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3ObjectMapper.java

```diff
@@ -48,13 +48,16 @@ public class S3ObjectMapper {
 
   private S3ObjectMapper() {}
 
-  static ObjectMapper mapper() {
+  public static ObjectMapper mapper() {
     if (!isInitialized) {
       synchronized (S3ObjectMapper.class) {
         if (!isInitialized) {
           MAPPER.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
           MAPPER.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-          MAPPER.setPropertyNamingStrategy(PropertyNamingStrategies.KebabCaseStrategy.INSTANCE);
+          // even though using new PropertyNamingStrategy.KebabCaseStrategy() is deprecated
+          // and PropertyNamingStrategies.KebabCaseStrategy.INSTANCE (introduced in jackson 1.14) is
+          // recommended, we can't use it because Spark still relies on jackson 1.13.x stuff
```

Comment:
Do we mean Jackson `2.13` and `2.14` instead of `1.x` here? I think this is okay but ideally what we do is we upgrade Iceberg Spark jackson itself then we can use the non-deprecated version.

Reply:
good catch, yes that should be 2.13.x